packages/portfolio-deploy/test/portfolio.test.ts

```diff
@@ -342,4 +349,88 @@ test.skip('open a USDN position', async t => {
   });
 });
 
+// Expect it fail when run independently
+test.serial('restart contract', async t => {
+  const {
+    runUtils: { EV },
+    agoricNamesRemotes,
+    walletFactoryDriver: wfd,
+  } = t.context;
+
+  t.truthy(agoricNamesRemotes.instance.ymax0);
+
+  const kit = await (EV.vat('bootstrap').consumeItem as ConsumeBootstrapItem)(
+    'ymax0Kit',
+  );
+  const actual = await EV(kit.adminFacet).restartContract(kit.privateArgs);
+
+  // Expect incarnation 1: first restart from initial deployment
+  // (The "remove old contract; start new contract" test creates a new contract
+  // instance, not an incarnation)
+  t.deepEqual(actual, { incarnationNumber: 1 });
+
+  // Test opening a portfolio after restart
+  for (const { msg, ack } of Object.values(
+    makeUSDNIBCTraffic('agoric1trader1', `${3_333 * 1_000_000}`),
+  )) {
+    protoMsgMockMap[msg] = ack;
+  }
+
+  const myMarshaller = makeClientMarshaller(v => (v as any).getBoardId());
+  const wallet = await wfd.provideSmartWallet(beneficiary, myMarshaller);
+
+  const { USDC, PoC26, BLD } = agoricNamesRemotes.brand as unknown as Record<
+    string,
+    Brand<'nat'>
+  >;
+  const give = harden({
+    Deposit: make(USDC, 3_333n * 1_000_000n),
+    Access: make(PoC26, 1n),
+    GmpFee: make(BLD, 1000n),
+  });
+
+  const ps = makeProposalShapes(USDC, BLD, PoC26);
+  mustMatch(harden({ give, want: {} }), ps.openPortfolio);
+
+  // Count portfolios before making the offer
+  const { storage } = t.context;
+
+  // XXX There is got to be a cleaner way to do this
```

Comment:
some improvement is using `readPublished`. I've made the change and pushed it. also fixed up the commit history so the ChainHub change isn't called a Ymax restart feature.